yt_dlp/extractor/mediasite.py

```diff
@@ -122,6 +122,57 @@ def _extract_urls(webpage):
                 r'(?xi)<iframe\b[^>]+\bsrc=(["\'])(?P<url>(?:(?:https?:)?//[^/]+)?/Mediasite/Play/%s(?:\?.*?)?)\1' % _ID_RE,
                 webpage)]
 
+    def __extract_slides(self, stream_id, snum, Stream, duration, images):
+        slide_base_url = Stream['SlideBaseUrl']
+
+        fname_template = Stream['SlideImageFileNameTemplate']
+        if fname_template != 'slide_{0:D4}.jpg':
+            self.report_warning('Unusual slide file name template; report a bug if slide downloading fails')
+        fname_template = re.sub(r'\{0:D([0-9]+)\}', r'{0:0\1}', fname_template)
+
+        fragments = []
+        for i, slide in enumerate(Stream['Slides']):
+            if i == 0:
+                if slide['Time'] > 0:
+                    default_slide = images.get('DefaultSlide')
+                    if default_slide is None:
+                        default_slide = images.get('DefaultStreamImage')
+                    if default_slide is not None:
+                        default_slide = default_slide['ImageFilename']
+                    if default_slide is not None:
+                        fragments.append({
+                            'path': default_slide,
+                            'duration': slide['Time'] / 1000,
+                        })
+
+            try:
+                next_time = Stream['Slides'][i + 1]['Time']
+            except IndexError:
```

Comment:
use `try_get`

Reply:
Now that I am rewriting this, this is what I came up with:

```python
            next_time = try_get(None, [
                lambda _: Stream['Slides'][i + 1]['Time'],
                lambda _: duration,
                lambda _: slide['Time'],
            ], expected_type=(int, float))
```

Is it worth the trouble?

(Ah, the joys of not having a generic Option monad.)

Reply:
When I wrote the comment, I was only thinking of changing the `try...except` to `try_get`, but this is fine as well